plugins/calibre.koplugin/wireless.lua

```diff
@@ -25,6 +26,58 @@ local T = FFIUtil.template
 
 require("ffi/zeromq_h")
 
+-- calibre broadcast ports used to find calibre server
+local BROADCAST_PORTS = {54982, 48123, 39001, 44044, 59678}
+-- calibre companion local port
+local COMPANION_PORT = 8134
+-- requests opcodes
+local OPCODES = {
+    OK                        = 0,
+    SET_CALIBRE_DEVICE_INFO   = 1,
+    SET_CALIBRE_DEVICE_NAME   = 2,
+    GET_DEVICE_INFORMATION    = 3,
+    TOTAL_SPACE               = 4,
+    FREE_SPACE                = 5,
+    GET_BOOK_COUNT            = 6,
+    SEND_BOOKLISTS            = 7,
+    SEND_BOOK                 = 8,
+    GET_INITIALIZATION_INFO   = 9,
+    BOOK_DONE                 = 11,
+    NOOP                      = 12,
+    DELETE_BOOK               = 13,
+    GET_BOOK_FILE_SEGMENT     = 14,
+    GET_BOOK_METADATA         = 15,
+    SEND_BOOK_METADATA        = 16,
+    DISPLAY_MESSAGE           = 17,
+    CALIBRE_BUSY              = 18,
+    SET_LIBRARY_INFO          = 19,
+    ERROR                     = 20,
+}
+
+-- Mark some strings for translation.
+-- luacheck: push ignore 511
+if false then
+    -- @translators tcp socket error on host:port
+    _("no route to host")
+    -- @translators tcp socket error on host:port
+    _("host not found")
+    -- @translators tcp socket error on host:port
+    _("connection refused")
+    -- @translators tcp socket error on host:port
+    _("timeout")
+    -- @translators tcp socket error on host:port
+    _("closed")
+    -- @translators calibre server address type
+    _("discovered")
+    -- @translators calibre server address type
+    _("specified")
+    -- @translators calibre connection error
+    _("handshake timeout")
+    -- @translators calibre connection error
+    _("invalid password")
+end
```

Comment:
You could also put it in a separate file that isn't required anywhere (though I suppose that has a larger chance of being lost).

Reply:
Yeah, since the only user is the calibre plugin, I think it's better to keep it here for now.

Reply:
Sounds good.